test/src/test/java/org/corfudb/runtime/object/transactions/OptimisticTransactionContextTest.java

```diff
@@ -28,6 +33,37 @@
     public void TXBegin() { OptimisticTXBegin(); }
 
 
+    @Test
+    public void testMultiStreamQuery() throws Exception {
+
+        CorfuRuntime rt = getDefaultRuntime();
+        rt.getParameters().setEnableMultiStreamQuery(true);
+
+        UUID streamId = UUID.randomUUID();
+        UUID streamId2 = UUID.randomUUID();
+
+        // increment the tail for streamId2
+        rt.getSequencerView().nextToken(Collections.singleton(streamId2), 2);
+
+        TXBegin();
+
+        TokenResponse resp = rt.getSequencerView().nextToken(Collections.singleton(streamId), 0);
+        // compute the snapshot ts
+        TransactionalContext.getCurrentContext().getSnapshotTimestamp();
+
+        // Drop all messages to the sequencer, this is used to verify that the stream tail queries
```

Comment:
This test appears to be overspecified. It should just test that a multistream query works and returns the correct information. 

If someone down the line changes the behavior of the system to have background RPCs to the sequencer, this test will break, even though multi-stream queries are not broken.